lib/cloud/clients.go

```diff
@@ -942,3 +1058,90 @@ func (c *TestCloudClients) GetAzureRunCommandClient(subscription string) (azure.
 func (c *TestCloudClients) Close() error {
 	return nil
 }
+
+// AWSSessionCacheKeyBuilder is a helper struct wrapper around a string builder
+// that is used to efficiently build cache keys for a given region and, optionally,
+// a chain of AWS assumed-role sessions.
+type AWSSessionCacheKeyBuilder struct {
+	sb        *strings.Builder
+	roleCount int
+}
+
+// NewAWSSessionCacheKeyBuilder constructs a new cache key builder intialized
+// with a specific AWS region.
+func NewAWSSessionCacheKeyBuilder(region string) AWSSessionCacheKeyBuilder {
+	sb := strings.Builder{}
+	_, _ = sb.WriteString(region) // infalliable.
```

Comment:
Still typo is present. I'm marking this as unresolved.

Reply:
True, I marked resolved because I'm going to remove this cache key builder, just didn't have time to do it yesterday. I'll resolve this once I push the change

Reply:
removed in fc800e15f75f7ed8f38b88b6aba00a756c66d144